kafka-rest/src/test/java/io/confluent/kafkarest/README.md

```diff
@@ -0,0 +1,34 @@
+Kafka REST Proxy Tests
+================
+
+Unit Tests
+----------
+
+The unit tests can be run using `mvn test`
+
+To run an individual unit test class use, for example, `mvn -Dtest=ProduceRequestTest test`
+
+To run an individual unit test within a test class use, for example, `mvn -Dtest=ProduceRequestTest#testProduceRequestDeserializerAddsSizeFromString test`
+
+Integration Tests
+-----------------
+
+The integration tests live in the integration folder.
+
+They can be run using `mvn failsafe:integration-test`
+
+Individual test classes can be run using, for example, `mvn -Dit.test=ProduceActionIntegrationTest failsafe:integration-test`
+
+And individual tests within a test class can be run using, for example, ` mvn -Dit.test=ProduceActionIntegrationTest#produceBinary failsafe:integration-test`
+
+Notes
+-----
+
+To run individual tests you must be within the kafka-rest project folder (eg `~/git/kafka-rest/kafka-rest`) otherwise no tests will be found.
+
+If you make changes to the code, you will need to rebuild before running the tests for any code changes to be picked up. For example rebuild with `mvn package -DskipTests=true`
```

Comment:
nit: Without checking what exactly might be causing a problem and what not, I'd err on the side of safety and recommend to also include cleaning - `mvn clean package -DskipTests` instead.